zerver/models.py

```diff
@@ -973,8 +973,8 @@ def clean(self) -> None:
         # this regex will incorrectly reject patterns that attempt to
         # escape % using %%.
         found_group_set: Set[str] = set()
-        group_match_regex = r"(?<!%)%\((?P<group_name>[^()]+)\)s"
-        for m in re.finditer(group_match_regex, self.url_format_string):
+        group_match_regex = re.compile(r"(?<!%)%\((?P<group_name>[^()]+)\)s")
+        for m in group_match_regex.finditer(self.url_format_string):
```

Comment:
It doesn't do anything to compile this inside a function -- it'll be recompiled every time you call it.  It could be reasonable to make this something compiled when the class is initialized.